pkg/nto/performanceprofile_test.go

```diff
@@ -0,0 +1,662 @@
+		testBuilder := NewBuilder(testSettings,
+			testCase.name,
+			testCase.cpuIsolated,
+			testCase.cpuReserved,
+			testCase.nodeSelector)
+		assert.Equal(t, testCase.expectedError, testBuilder.errorMsg)
+		assert.NotNil(t, testBuilder.Definition)
+
+		if testCase.expectedError == "" {
+			assert.Equal(t, testCase.name, testBuilder.Definition.Name)
+		}
+	}
+}
+
+func TestPerformanceProfileExists(t *testing.T) {
+	testCases := []struct {
+		testPerformanceProfile *Builder
+		expectedStatus         bool
+	}{
+		{
+			testPerformanceProfile: buildValidPerformanceProfileBuilder(buildPerformanceProfileWithDummyObject()),
+			expectedStatus:         true,
+		},
+		{
+			testPerformanceProfile: buildInValidPerformanceProfileBuilder(buildPerformanceProfileWithDummyObject()),
+			expectedStatus:         false,
+		},
+		{
+			testPerformanceProfile: buildValidPerformanceProfileBuilder(clients.GetTestClients(clients.TestClientParams{})),
+			expectedStatus:         false,
+		},
+	}
+
+	for _, testCase := range testCases {
+		exist := testCase.testPerformanceProfile.Exists()
+		assert.Equal(t, testCase.expectedStatus, exist)
+	}
+}
+
+func TestPerformanceProfileGet(t *testing.T) {
+	testCases := []struct {
+		testPerformanceProfile *Builder
+		expectedError          error
+	}{
+		{
+			testPerformanceProfile: buildValidPerformanceProfileBuilder(buildPerformanceProfileWithDummyObject()),
+			expectedError:          nil,
+		},
+		{
+			testPerformanceProfile: buildInValidPerformanceProfileBuilder(buildPerformanceProfileWithDummyObject()),
+			expectedError:          fmt.Errorf("PerformanceProfile's name is empty"),
+		},
+		{
+			testPerformanceProfile: buildValidPerformanceProfileBuilder(clients.GetTestClients(clients.TestClientParams{})),
+			expectedError:          fmt.Errorf("performanceprofiles.performance.openshift.io \"default\" not found"),
+		},
+	}
+
+	for _, testCase := range testCases {
+		performanceProfileObj, err := testCase.testPerformanceProfile.Get()
+
+		if testCase.expectedError == nil {
+			assert.Equal(t, performanceProfileObj.Name, testCase.testPerformanceProfile.Definition.Name)
+			assert.Nil(t, err)
+		} else {
+			assert.Equal(t, testCase.expectedError.Error(), err.Error())
+		}
+	}
+}
+
+func TestPerformanceProfileCreate(t *testing.T) {
+	testCases := []struct {
+		testPerformanceProfile *Builder
+		expectedError          string
+	}{
+		{
+			testPerformanceProfile: buildValidPerformanceProfileBuilder(buildPerformanceProfileWithDummyObject()),
+			expectedError:          "",
+		},
+		{
+			testPerformanceProfile: buildInValidPerformanceProfileBuilder(buildPerformanceProfileWithDummyObject()),
+			expectedError:          "PerformanceProfile's name is empty",
+		},
+		{
+			testPerformanceProfile: buildValidPerformanceProfileBuilder(clients.GetTestClients(clients.TestClientParams{})),
+			expectedError:          "resourceVersion can not be set for Create requests",
```

Comment:
This test case should't return error. It has to create object successfully. You just need to remove resourceVersion from structure